CIP-????/README.md

```diff
@@ -0,0 +1,119 @@
+
+2. Sometimes, the endpoint ```url``` field is not enough to determine its purpose. The user should understand perfectly the objective of the payload which he or she is signing. That's why the payload MUST contain an ```action``` field with a descriptive text containing the purpose of the payload. 
+
+3. The payload MUST include a UNIX timestamp. The ```timestamp``` field is used as a nonce and a mark for the payload expiration. This field is very imporant in case the payload is compromised.
+
+Additional fields MAY be included in the payload. Depending on the process, it could be useful to include some aditional fields in the protected header of the signature. For example, the email information should be included in the protected header in a registration request. In that way, this payload can be used only to register that specific email and it can not be tampered. 
+
+```JSON
+{
+    "url": "http://example.com/signup",
+    "action": "Sign up",
+    "timestamp": 1673261248,
+    ["email": "email@example.com"] <- Optional
+}
+```
+
+### Wallet specification
+
+The wallets can improve the overall security implementing the following guidelines. We RECOMMEND to show in a structured way the payload information for shake of clarity. This information should be well understanded by the users before the payload is signed.  
+
+The ```url``` field provides information about the hostname of the application. This hostname MUST be included in the wallet allow list. If a known domain A tries to sign a payload for an unknown domain B, you will be prompted with permission popup making more obvious the cross-domain interaction. When possible, the wallet SHOULD warn the user if a payload is for a different domain.
+
+The wallet SHOULD update the ```timestamp``` field to the current time just before the signature. This field ideally should match the moment just before the signature, thereby the server recieves the more updated payload possible. 
+
+### Server processing
+
+The server has ultimate responsibility of processing correctly the requests. We use the content to validate the payload. The request will be processed with the following steps:
+
+1. The server MUST check the action and the endpoint included in the request. Each route to an endpoint MUST have an associated action and a URL. The first step is to check that they match with the parameterized action.
+
+2. The server MUST check the expiration of the payload. The expiration SHOULD be enough to give time to the user to introduce the wallet password but it SHOULD NOT be too long, we RECOMMEND not more than 5 minutes.
+
+3. The server MUST validate the COSESign1 signature and check that the address inside the protected map of the signature corresponds to the public key in the COSEKey. 
+
+Additionally the server COULD extract the payload content and pass it through the server logic.
+
+## Rationale: how does this CIP achieve its goals?
```

Comment:
One question I'd like to see answered in the 'Rationale' is what value does such an authorization has from a service provider standpoint (assumed to be 'the server' here)? Said differently, why would my-awesome-dapp care that a user authorized something with its wallet? Is the service provider assumed to have some knowledge of the key involved in the authorization and is only expecting a proof of ownership? How is such a proof better than a transaction witness proving the ability to spend associated UTxOs?

Reply:
@KtorZ, I made some improvement in the Rationale. I think now it's a little bit more specific.

Reply:
@jmagan what's currently in the Rationale is more specific, with respect to currently available alternatives that are trying to do the same thing, but it doesn't go into as much detail as @KtorZ is requesting above.  If it answers the question about "How does the dApp trust the wallet in this method?" then can you please explain it here and/or in the CIP?  (@KtorZ please mark Resolved if you think it is)

Reply:
@rphair  Okey I see the point. I though that this point would clarify it: 

>  Moreover, the signature method proposed in this CIP does not require users to spend funds in a transaction, which further lowers the cost and barriers to entry for users.

In relation to your question 

>How does the dApp trust the wallet in this method?

I can add a brief summary of or a referenfe to the CIP-0008 but I though It could be a little bit redundant.

Reply:
I also thought that sentence was relevant to the question: so maybe a bit more detail, maybe even a subheading on that subject, would make the advantage more apparent to service providers... since generally they're the ones who will be implementing your proposal 🤓

Reply:
I made a connection between CIP-0008 and this CIP at the beggining of the Rationale. I feel that previously this part was missing and now all make more sense.

> ## Rationale: how does this CIP achieve its goals?
> 
> CIP-0008 enhances authentication by enabling individuals to prove ownership of addresses, identities, and off-chain data through message signing. It provides a reliable means of authentication by allowing individuals to attach a public key to their data and sign messages associated with that data, thereby establishing ownership and ensuring the integrity of the authentication process.
> 
> Additionally, This specification provides the general guidelines and necessary recommendations for performing secure authenticated web3 requests in the Cardano ecosystem. It covers the two main desired characteristics for a secure payload: It must expire and it must be non-static. Moreover, the signature method proposed in this CIP does not require users to spend funds in a transaction, which further lowers the cost and barriers to entry for users.